### ISSUE TITLE
Incorrect handling of compilation flags for xrootd Python bindings?

### ISSUE THREAD

**Issue by dkonst13** [xrootd/xrootd#2001]
Dear @amadio, experts, 

We encountered an issue building xrootd for MacOS platforms. We discovered that no C++ standard was being passed when compiling xrootd Python bindings.

After investigating, we found that redefining cfg_vars["OPT"], cfg_vars["CFLAGS"], and cfg_vars["PY_CFLAGS"] in setup.py did not actually change the compilation flags. We attempted to append some meaningless flags to them, and it did not fail for Linux platforms. It proves that it is not propagated properly.

why not use the approach outlined in this resource on exploiting extra_compile_args?

https://pythonextensionpatterns.readthedocs.io/en/latest/compiler_flags.html

```
extra_compile_args += ["-std=c++11", "-Wall", "-Wextra"]
if _DEBUG:
    extra_compile_args += ["-g3", "-O0", "-DDEBUG=%s" % _DEBUG_LEVEL, "-UNDEBUG"]
else:
    extra_compile_args += ["-DNDEBUG", "-O3"]
```

**Comment by amadio** [comment]
Hi @dkonst13. We are aware of this (#1807). The Python packaging will be rewritten using latest tools to address this and other problems (see #1844).

**Comment by amadio** [comment]
Fixed by #2025.
